### ISSUE TITLE
CI: S3 source is failing

### ISSUE THREAD

**Issue by vdesabou** [vdesabou/kafka-docker-playground#184]
https://github.com/vdesabou/kafka-docker-playground/runs/1695026803?check_suite_focus=true

```
[2021-01-13 12:59:55,391] ERROR WorkerConnector{id=s3-source} Error while starting connector (org.apache.kafka.connect. runtime. WorkerConnector)
java. lang. NullPointerException
at io.confluent.connect.s3. source.S3Storage.newS3Client (S3Storage. java:409)
at io.confluent.connect.s3. source. S3Storage.<init>(S3Storage. java: 63)
at io.confluent.connect.s3.source.S3SourceConnector.createStorage(S3SourceConnector.java:39)
at
io.confluent.connect.cloud.storage.source.StorageSourceConnector.start(StorageSourceConnector.java:65)
at org. apache. kafka. connect. runtime. WorkerConnector. doStart (WorkerConnector. java: 186)
at org.apache.kafka.connect. runtime. WorkerConnector. start(WorkerConnector. java: 211)
at org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:350)
at org.apache.kafka.connect. runtime. WorkerConnector. doTransitionTo (WorkerConnector. java: 333)
at org.apache.kafka.connect. runtime. WorkerConnector. doRun (WorkerConnector. java: 141)
at org. apache. kafka. connect. runtime. Worker Connector. run (WorkerConnector. java: 118)
at java. base/java.util. concurrent.Executors$RunnableAdapter.call(Executors. java:515)
at java. base/java. util. concurrent. FutureTask. run (FutureTask. java: 264)
at java. base/java. util. concurrent. ThreadPoolExecutor. runWorker (ThreadPoolExecutor. java: 1128)
at java. base/java.util. concurrent. ThreadPoolExecutor$Worker. run(ThreadPoolExecutor. java: 628)
at java. base/java. lang. Thread. run (Thread. java: 834)
[2021-01-13 12:59:55,396] ERROR [Worker clientId=connect-1, groupId=connect-cluster] Failed to start connector 's3-source'
(org.apache.kafka.connect.runtime.distributed.DistributedHerder)
org. apache. kafka. connect. errors. ConnectException: Failed to start connector: s3-source
at org.apache.kafka.connect.runtime.distributed.DistributedHerder.lambda$startConnector$5(DistributedHerder.java:1304)
at org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:336)
at org.apache.kafka.connect. runtime. WorkerConnector. doRun (WorkerConnector. java: 141)
at org.apache.kafka.connect. runtime. WorkerConnector.run(WorkerConnector. java: 118)
at java. base/java. util. concurrent. Executors$RunnableAdapter.call(Executors. java:515)
at java. base/java. util. concurrent. FutureTask. run (FutureTask. java: 264)
at java. base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java. base/java. util. concurrent. ThreadPoolExecutor$Worker. run(ThreadPoo lExecutor. java: 628)
at java. base/java. lang. Thread. run (Thread. java: 834)
```

**Comment by vdesabou** [comment]
`1.4.1` is failing with:

```
[2021-01-14 09:38:26,274] ERROR WorkerConnector{id=s3-source} Error while starting connector (org.apache.kafka.connect.runtime.WorkerConnector)
java.lang.NullPointerException
        at com.amazonaws.services.s3.internal.ServiceUtils.isS3AccelerateEndpoint(ServiceUtils.java:500)
        at com.amazonaws.services.s3.AmazonS3Client.setEndpoint(AmazonS3Client.java:722)
        at com.amazonaws.client.builder.AwsClientBuilder.setRegion(AwsClientBuilder.java:453)
        at com.amazonaws.client.builder.AwsClientBuilder.configureMutableProperties(AwsClientBuilder.java:424)
        at com.amazonaws.client.builder.AwsSyncClientBuilder.build(AwsSyncClientBuilder.java:46)
        at io.confluent.connect.s3.source.S3Storage.newS3Client(S3Storage.java:417)
        at io.confluent.connect.s3.source.S3Storage.<init>(S3Storage.java:63)
        at io.confluent.connect.s3.source.S3SourceConnector.createStorage(S3SourceConnector.java:39)
        at io.confluent.connect.cloud.storage.source.StorageSourceConnector.start(StorageSourceConnector.java:65)
        at org.apache.kafka.connect.runtime.WorkerConnector.doStart(WorkerConnector.java:186)
        at org.apache.kafka.connect.runtime.WorkerConnector.start(WorkerConnector.java:211)
        at org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:350)
        at org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:333)
        at org.apache.kafka.connect.runtime.WorkerConnector.doRun(WorkerConnector.java:141)
        at org.apache.kafka.connect.runtime.WorkerConnector.run(WorkerConnector.java:118)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
[2021-01-14 09:38:26,277] ERROR [Worker clientId=connect-1, groupId=connect-cluster] Failed to start connector 's3-source' (org.apache.kafka.connect.runtime.distributed.DistributedHerder)
org.apache.kafka.connect.errors.ConnectException: Failed to start connector: s3-source
        at org.apache.kafka.connect.runtime.distributed.DistributedHerder.lambda$startConnector$5(DistributedHerder.java:1304)
        at org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:336)
        at org.apache.kafka.connect.runtime.WorkerConnector.doRun(WorkerConnector.java:141)
        at org.apache.kafka.connect.runtime.WorkerConnector.run(WorkerConnector.java:118)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by vdesabou** [comment]
Fixed with `v1.4.2`
